src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1489,9 +1488,31 @@ public function getAutoconfiguredInstanceof(): array
     }
 
     /**
-     * @return array<string, callable>
+     * @return array<class-string, callable>
+     *
+     * @deprecated Use {@see getAttributeConfigurators()} instead
      */
     public function getAutoconfiguredAttributes(): array
+    {
+        trigger_deprecation('symfony/dependency-injection', '7.3', 'The "%s()" method is deprecated, use "getAttributeConfigurators()" instead.', __METHOD__);
+
+        return array_map(static function (array $configurators): callable {
+            if (count($configurators) === 1) {
+                return $configurators[0];
+            }
+
+            return static function (...$args) use ($configurators) {
```

Comment:
will this work when inspected by reflection?

Reply:
No, but the compiler pass uses the initial list of callbacks that is returned by the new `getAttributeConfigurators()` method instead of this.
This is only for backward compatibility if someone calls this method and calls the callable.

Reply:
what about throwing instead? wouldn't this mimick the previous exception close enough?

Reply:
SGTM. PR updated.